In [13]:
import pandas as pd
import numpy as np

In [14]:
df = pd.read_csv('Base de Datos.csv', delimiter=';')

In [15]:
df.head()

AGNO  RBD  DGV_RBD                              NOM_RBD  COD_REG_RBD  \
0  2023    1        9              LICEO POLITECNICO ARICA           15   
1  2023    2        7            PARVULARIO LAS ESPIGUITAS           15   
2  2023    3        5  ESC. PEDRO VICENTE GUTIERREZ TORRES           15   
3  2023    4        3            LICEO OCTAVIO PALMA PEREZ           15   
4  2023    5        1             JOVINA NARANJO FERNANDEZ           15   

  NOM_REG_RBD_A  COD_PRO_RBD  COD_COM_RBD NOM_COM_RBD  COD_DEPROV_RBD  ...  \
0           AYP          151        15101       ARICA             151  ...   
1           AYP          151        15101       ARICA             151  ...   
2           AYP          151        15101       ARICA             151  ...   
3           AYP          151        15101       ARICA             151  ...   
4           AYP          151        15101       ARICA             151  ...   

  DC_SUP_SOST  HH_SUP_SOST  DC_SUBDIR  HH_SUBDIR  DC_PROF_ENC  HH_PROF_ENC  \
0           0            0          0          0            0            0   
1           0            0          0          0            0            0   
2           0            0          0          0            0            0   
3           0            0          0          0            0            0   
4           0            0          0          0            0            0   

   DC_EDUC_TRAD  HH_EDUC_TRAD  DC_TOT  HH_TOT  
0             0             0      71    2393  
1             0             0      15     530  
2             0             0      27    1078  
3             0             0      69    2727  
4             0             0      63    2414  

[5 rows x 47 columns]

In [16]:
df.columns

Index(['AGNO', 'RBD', 'DGV_RBD', 'NOM_RBD', 'COD_REG_RBD', 'NOM_REG_RBD_A',
       'COD_PRO_RBD', 'COD_COM_RBD', 'NOM_COM_RBD', 'COD_DEPROV_RBD',
       'NOM_DEPROV_RBD', 'COD_DEPE', 'COD_DEPE2', 'RURAL_RBD', 'ESTADO_ESTAB',
       'DC_A', 'HH_A', 'DC_UTP', 'HH_UTP', 'DC_PDIR', 'HH_PDIR', 'DC_DIR',
       'HH_DIR', 'DC_OES', 'HH_OES', 'DC_OF', 'HH_OF', 'DC_JUTP', 'HH_JUTP',
       'DC_IG', 'HH_IG', 'DC_OR', 'HH_OR', 'DC_DIR_SOST', 'HH_DIR_SOST',
       'DC_TP_SOST', 'HH_TP_SOST', 'DC_SUP_SOST', 'HH_SUP_SOST', 'DC_SUBDIR',
       'HH_SUBDIR', 'DC_PROF_ENC', 'HH_PROF_ENC', 'DC_EDUC_TRAD',
       'HH_EDUC_TRAD', 'DC_TOT', 'HH_TOT'],
      dtype='object')

In [17]:
df.describe()

AGNO           RBD       DGV_RBD   COD_REG_RBD   COD_PRO_RBD  \
count  16654.0  16654.000000  16654.000000  16654.000000  16654.000000   
mean    2023.0  13463.826768      4.508767      9.245407     94.462231   
std        0.0  10676.369828      2.872668      3.813839     37.829669   
min     2023.0      1.000000      0.000000      1.000000     11.000000   
25%     2023.0   5012.500000      2.000000      6.000000     61.000000   
50%     2023.0  10443.500000      5.000000      9.000000     91.000000   
75%     2023.0  20328.750000      7.000000     13.000000    131.000000   
max     2023.0  42192.000000      9.000000     16.000000    163.000000   

        COD_COM_RBD  COD_DEPROV_RBD      COD_DEPE     COD_DEPE2     RURAL_RBD  \
count  16654.000000    16654.000000  16654.000000  16654.000000  16654.000000   
mean    9452.490393       91.813919      2.813859      1.884532      0.295905   
std     3785.177284       36.277812      1.026080      0.928506      0.456462   
min     1101.000000       11.000000      1.000000      1.000000      0.000000   
25%     6116.000000       61.000000      2.000000      1.000000      0.000000   
50%     9117.000000       92.000000      3.000000      2.000000      0.000000   
75%    13119.000000      133.000000      3.000000      2.000000      1.000000   
max    16305.000000      151.000000      6.000000      5.000000      1.000000   

       ...   DC_SUP_SOST   HH_SUP_SOST     DC_SUBDIR     HH_SUBDIR  \
count  ...  16654.000000  16654.000000  16654.000000  16654.000000   
mean   ...      0.005764      0.213042      0.048397      2.024018   
std    ...      0.082537      3.277349      0.249541     10.606641   
min    ...      0.000000      0.000000      0.000000      0.000000   
25%    ...      0.000000      0.000000      0.000000      0.000000   
50%    ...      0.000000      0.000000      0.000000      0.000000   
75%    ...      0.000000      0.000000      0.000000      0.000000   
max    ...      2.000000     88.000000      7.000000    289.000000   

        DC_PROF_ENC   HH_PROF_ENC  DC_EDUC_TRAD  HH_EDUC_TRAD        DC_TOT  \
count  16654.000000  16654.000000  16654.000000  16654.000000  16654.000000   
mean       0.123394      5.352408      0.041852      1.162243     16.746607   
std        0.328899     14.313592      0.275703      9.611569     23.108197   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        0.000000      0.000000      0.000000      0.000000      0.000000   
50%        0.000000      0.000000      0.000000      0.000000      6.000000   
75%        0.000000      0.000000      0.000000      0.000000     26.000000   
max        1.000000     44.000000     13.000000    548.000000    255.000000   

             HH_TOT  
count  16654.000000  
mean     625.700492  
std      892.985992  
min        0.000000  
25%        0.000000  
50%      219.000000  
75%      950.000000  
max    11192.000000  

[8 rows x 43 columns]

In [18]:
df.isnull().sum()

AGNO               0
RBD                0
DGV_RBD            0
NOM_RBD           30
COD_REG_RBD        0
NOM_REG_RBD_A      0
COD_PRO_RBD        0
COD_COM_RBD        0
NOM_COM_RBD        0
COD_DEPROV_RBD     0
NOM_DEPROV_RBD     0
COD_DEPE           0
COD_DEPE2          0
RURAL_RBD          0
ESTADO_ESTAB       0
DC_A               0
HH_A               0
DC_UTP             0
HH_UTP             0
DC_PDIR            0
HH_PDIR            0
DC_DIR             0
HH_DIR             0
DC_OES             0
HH_OES             0
DC_OF              0
HH_OF              0
DC_JUTP            0
HH_JUTP            0
DC_IG              0
HH_IG              0
DC_OR              0
HH_OR              0
DC_DIR_SOST        0
HH_DIR_SOST        0
DC_TP_SOST         0
HH_TP_SOST         0
DC_SUP_SOST        0
HH_SUP_SOST        0
DC_SUBDIR          0
HH_SUBDIR          0
DC_PROF_ENC        0
HH_PROF_ENC        0
DC_EDUC_TRAD       0
HH_EDUC_TRAD       0
DC_TOT             0
HH_TOT             0
dtype: int64

**3.1 Pregunta 0 (descripción de variables)**

Con el objetivo de familiarizarse y entender mejor la base de datos, se requiere realizar un analisis de esta.

1. Generar un grafico por cada una de las variables. El grafico debe tener un tıtulo descriptivo y debe mostrar
informacion clara de la variable.

2. Se debe indicar por cada una de las variables su tipo (cuantitativa o cualitativa) justificando su decision. Se
debe recordar que el tipo de variable depende de su naturaleza, no del formato en el que se encuentre en la
base de datos.

**3.2 Pregunta 1 (test de independencia)**

Los docentes UTP (Unidad Tecnico Pedagogica) en Chile son profesionales de la educacion encargados de la coordinacion pedagogica y tecnica en las instituciones educativas. Se dedican a la planificacion, evaluacion y seguimiento del rendimiento academico de los estudiantes, ofrecen apoyo y orientacion a otros docentes, participan en el desarrollo profesional y gestionan aspectos curriculares. Tambien mantienen comunicacion con los padres para informarles sobre el progreso de los estudiantes, desempeñando un papel fundamental en la mejora de la calidad educativa en
el paıs.

1. Obtenga un grafico apropiado que le permita estudiar la distribucion del numero de docentes pertenecientes a
la planta UTP en los establecimientos educacionales de sus regiones. Discuta las caracterısticas m´as relevantes
presentes en sus graficos

2. Al Ministerio de Educacion le preocupa que existan diferencias en el numero de docentes UTP entre los distintos tipos de establecimientos (municipales, particulares subvencionados y particulares pagados). Implemente un test de independencia chi-cuadrado que le permita concluir si existe dependencia entre el tipo de establecimiento: (solo considere municipales, particulares subvencionados y particulares pagados) y el numero
de docentes UTP. En particular:

• Reporte la tabla de frecuencias observadas y esperadas.

• Reporte el p-valor obtenido y concluya con un valor α = 0.05. Interprete sus resultados.

3. Obtenga un grafico apropiado para mostrar la distribucion del numero de docentes de planta UTP por establecimiento. Finalmente, muestre como esa distribucion cambia (o no) para los distintos establecimientos: Municipales, particulares suvencionados y particulares pagados. Incluya una breve discusion sobre si lo que observa en sus graficos concuerda con lo obtenido en la implementacion del test anterior.

**3.3 Pregunta 2 (test de bondad de ajuste)**

En Chile, las “Horas de Contrato” (variable HH_A) para los docentes de aula se refieren a la cantidad de horas que los
profesores est´an contratados para trabajar en una institucion educativa durante un perıodo especıfico, generalmente
un año escolar. La cantidad de horas de contrato puede variar segun el nivel educativo y el tipo de establecimiento
educacional.
En el contexto de un estudio educativo, se plantea la hipotesis de que las Horas de Contrato de los Docentes
de Aula siguen una distribucion exponencial. Para investigar esta afirmacion, sigua los siguientes pasos:

1. Analisis Visual: Genere un histograma de densidad para la variable en cuestion y añada en este grafico la funcion de densidad asociada a la distribucion exponencial. Disculta sobre la factibilidad de la hipotesis planteada en el estudio utilizando el histograma generado.

**Hint**: Recuerde que la densidad de la distribucion exponencial esta dada por:
f(x) = λ exp{−λx},
y que para la construccion de su grafico es razonable estimar el parametro λ usando X¯ −1 donde X¯ representa el promedio de HH_A.

2. Evaluacion Estadıstica: Implemente un test de hipotesis de bondad de ajuste con un nivel de significancia de α = 0.05 que le permita concluir sobre la hipotesis planteada en el estudio. Recuerde que el test de bondad
de ajuste le permite analizar si los datos se comportan como una distribucion en especıfico.

3.4 Pregunta 3 (regresion lineal simple y multiple)

1. Regresion lineal simple: Implemente un modelo de regresion lineal simple que le permita modelar las
Horas de Contrato de los Docentes de Aula (variable HH_A) en funcion del Total de Docentes de aula (variable
DC_A).
Los pasos que debera realizar para dar respuesta a esta pregunta son:

(a) Obtenga la correlacion entre las variables involucradas.

(b) Visualize la relacion entre las variables

(c) Separe los datos en conjuntos de datos de entrenamiento y testeo. Luego ajuste el modelo de regresion.

(d) Muestre en una tabla los coeficientes obtenidos (Intercepto y pendiente). Interprete sus resultados.

(e) Visualize el modelo de regresion utilizando los coeficientes encontrados en el paso anterior.

(f) Obtenga y comente el coeficiente de determinacion obtenido.

(g) Evalue su modelo utilizando los datos del set de testeo. 
Para esto, calcule el MSE y RMSE predictivos
(error cuadratico medio y raız del error cuadratico medio). ¿Que significan los valores obtenidos?

2. Regresion lineal multiple: En esta pregunta se le pide que implemente un modelo de regresion lineal
multiple para modelar las Horas de Contrato de los Docentes de Aula (variable HH_A) en funcion del Total
de Docentes de aula (variable DC_A) y de tres variables mas que usted debe escoger. Para escoger
estas variables usted debe usar criterios claros y objetivos (por ejemplo, puede elegir las variables que mas se
correlacionan con la variable dependiente).

Utilizando la misma separacion de entrenamiento y testing realizada en el ıtem anterior responda: ¿Que
impacto tienen las variable adicionales en el rendimiento de su modelo? Comente en funcion de los nuevas
metricas obtenidas (error cuadratico medio y R2).

**3.5 Pregunta 4 (clasificacion y validacion)**

En esta pregunta se le pide que implemente dos modelos de clasificacion: Support Vector Machine y Regresion
Logıstica. El objetivo es generar un modelo que prediga si la variable Horas de Contrato de los Docentes de Aula
(variable HH A) es mayor a 160 horas. Los modelos deben ser implementados usando validacion simple y K-Fold
Cross validation. Los pasos para responder la pregunta son:

1. Genere una nueva variable a partir de Horas de Contrato de los Docentes de Aula (variable HH A) que tome
el valor de 1 si supera las 160 horas, y tenga valor 0 si es menor o igual a este valor.

2. Seleccione la variable Total de Docentes de aula (variable DC A) y las tres mejores variables. Para escoger
estas variables usted debe justificar usando criterios claros y objetivos.

3. Genere un nuevo dataframe que contenga: la variable Total de Docentes de aula, las 3 variables que escogio
y la nueva variable generada

4. Con el data frame obtenido, aplique el modelo SVM utilizando la tecnica de validacion simple. Obtenga los
valores de las metricas: accuracy, recall, precision y f1 score. Explique como interpretarıa los resultados de
estas metricas.

5. Con el data frame obtenido, aplique el modelo Regresion Logıstica utilizando la tecnica de validacion simple.
Obtenga los valores de las metricas: accuracy, recall, precision y f1 score. Explique como interpretarıa los
resultados de estas metricas.

6. Mencione la principal deficiencia de aplicar validacion simple aplicada para este caso.

7. Con el data frame obtenido, aplique el modelo SVM utilizando la tecnica de validacion cruzada. Obtenga
los valores de las metricas: accuracy, recall, precision y f1 score. Determine si su modelo se encuentra sobre
entrenado.

8. Con el data frame obtenido, aplique el modelo Regresion Logıstica utilizando la tecnica de validacion cruzada.
Obtenga los valores de las metricas: accuracy, recall, precision y f1 score. Determine si su modelo se encuentra
sobre entrenado.